### PR TITLE
Add new backgrounds (night, dynamic)

### PR DESCRIPTION
**Pull request by RedSparr0w**
Night background (thanks to @LuqDragon)
image

Dynamic Background:
(changes depending on time of day, pokemon walk/fly across the background occasionally)
_background images from https://bitday.me/, and code heavily modified from there_
Pokémon sprites from reddit, Unknown original source
image
Example of just the background (sped up, with lots of Pokémon):
ezgif com-video-to-gif (2)

### CODE REVIEW

**Comment by Ishadijcks** [comment]
That's a lot of files, does it come with settings? (Night, day, dynamic)?

**Comment by RedSparr0w** [comment]
Yeah, a lot of Pokémon walking sprites 😅 
Sure does:
image

**Comment by Ishadijcks** [comment]
Nice!

Couldn't check on mobile :')

**Review by anapple96 — APPROVED** [review]

File: src/scripts/DynamicBackground.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    constructor() {}
+
+    /*
+    SUN & MOON
+    */
+    static setSunMoonPosition = (date = new Date()) => {
+        const w = window.innerWidth / 1.2;
+
+        // Do the same thing with the height. Responsive = Good times.
+        const h = window.innerHeight / 1;
```
Comment (Aegyo) [inline src/scripts/DynamicBackground.ts:11]: `/ 1` :thinking:

File: src/scripts/DynamicBackground.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+        // Get the hours and minutes.
+        const hours = date.getHours();
+        const mins = date.getMinutes();
+
+        // Calculate the position of the sun and moon based on the time.
+        const sunRad = (((hours) * 60 + mins) / (24.00 * 62.00)) * Math.PI * 2;
```
Comment (Aegyo) [inline src/scripts/DynamicBackground.ts:18]: This `62` looks like it should be `60`, and what is the purpose of the `.00` ?

File: src/scripts/DynamicBackground.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        const moonRad = (((hours + 12) * 60 + mins) / (24.00 * 60.00)) * Math.PI * 2;
+
+        // Calculate the axis
+        const sunX = (w / 1.8) - (w * Math.sin(sunRad)) / 2;
+        const sunY = (h / 2) + (h * Math.cos(sunRad)) / 2;
+        const moonX = (w / 1.8) - (w * Math.sin(moonRad)) / 2;
+        const moonY = (h / 1.4) + (h * Math.cos(moonRad)) / 2;
+
+
+        // Apply the poisitions based on our previous calculations
```
Comment (Aegyo) [inline src/scripts/DynamicBackground.ts:28]: `poisitions` spelling error

File: src/scripts/DynamicBackground.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    static updateScene = (date = new Date()) => {
+        try {
+            DynamicBackground.setSunMoonPosition(date);
+            DynamicBackground.updateBackgrounds(date);
+        } catch (e) {}
+    };
+
+    // Determines the images to use based on the hour
+    static getPicture = hour => {
```
Comment (Aegyo) [inline src/scripts/DynamicBackground.ts:74]: This is looking an awful lot like `floor((hour - 1) / 2)` with some fudging for hour < 1

File: src/scripts/DynamicBackground.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            document.getElementById('dynamic-background').removeChild(pokeElement);
+        }, GameConstants.MINUTE);
+    };
+
+    static active = false;
+
+    static startAddingPokemon = () => {
+        const delay = Math.floor(Math.random() * (10 * GameConstants.SECOND));
+        setTimeout(() => {
+            // todo update highest ID
```
Comment (Aegyo) [inline src/scripts/DynamicBackground.ts:127]: Any reason we are not doing this now?

File: src/scripts/settings/Settings.ts [hunk 6]
```diff
@@ -118,4 +126,29 @@ Settings.add(new MultipleChoiceSetting('partySort', 'Sort:',
...
 ));
-Settings.add(new BooleanSetting('partySortDirection', 'reverse', false));
\ No newline at end of file
+Settings.add(new BooleanSetting('partySortDirection', 'reverse', false));
+
+/*
+ * SUBSCRIBERS
+ */
+
+let autoUpdateScene;
```
Comment (Aegyo) [inline src/scripts/settings/Settings.ts:135]: Can we put this on the DynamicBackground class?

**Review by Aegyo — COMMENTED** [review]
A great addition.
Some minor questions about the code below.
Also, the moon is looking quite pixelated, it looks a bit strange on the Night background next to much higher definition pokemon images. Anything we can do about that?

**Comment by LuqDragon** [comment]
just here?
bug2

**Comment by LuqDragon** [comment]
i think is better to set the dynamic to night or day to simulate a dark/light background, just to standardize the things

File: src/scripts/DynamicBackground.ts (hunk 3 above)
Comment (RedSparr0w) [reply src/scripts/DynamicBackground.ts:28]: Hmm, someone must be messing with my code 😓

File: src/scripts/DynamicBackground.ts (hunk 5 above)
Comment (RedSparr0w) [reply src/scripts/DynamicBackground.ts]: Wasn't sure if we should limit it to Pokémon in the current region/highest region or just allow free reign.
Just gone for highest region though.

**Comment by RedSparr0w** [comment]
Also think I've also fixed the jitter, couldn't replicate it completely, but notice when you click the start menu it flashed briefly, something to do with the bootstrap handling of it.

**Review by Aegyo — APPROVED** [review]
I wasn't able to get the jittering to happen before, but I was getting a horizontal scrollbar which has since stopped happening. Might want to wait for @LuqDragon to confirm the jittering is gone though.

**Comment by LuqDragon** [comment]
i think that was a problem with my repo in my machine, the jittering is gone, but i found a "bug". I was not seeing the dynamic background in my regular browser, i change it and it works. So i realized that an extension named "Dark Reader" that a set a "dark theme" to all the sites is giving me the day background instead dynamic, this is so bizarre

**Pull request merged**